packages/client/hmi-client/src/components/responsive-matrix/pixi-utils.ts

```diff
@@ -52,7 +52,33 @@ export class CustomBufferResource extends Resource {
 
 		return true;
 	}
-}
+};
+
+/**
+ * Get the WebGL max texture size limit using the pixi application gl instance.
+ * @param {Application} app
+ */
+export const getGlMaxTextureSize = (app: Application): number => {
+	const { gl } = (app.renderer as any).context;
+	return gl.getParameter(gl.MAX_TEXTURE_SIZE);
+};
+
+/**
+ * Given a data size and a max texture size, find the dimensions of the smallest 2D texture required
+ * to hold the data.
+ * @param {number} dataSize
+ * @param {number} maxTextureSize
+ */
+export const getTextureDim = (dataSize: number, maxTextureSize: number) => {
+	const dataSizeToMaxRatio = dataSize / maxTextureSize;
+	const x = dataSizeToMaxRatio >= 1 ? maxTextureSize : dataSize % maxTextureSize || 1;
+	const y = Math.ceil(dataSizeToMaxRatio) || 1
+	return {
+		x,
+		y,
+		n: x * y,
+	};
+};
```

Comment:
Would be good if you can provide an example in the comment and explain what x/y/n are used for